core/unit_test/TestCreateMirror.cpp

```diff
@@ -0,0 +1,158 @@
+  check_memory_space(create_mirror_view(view_alloc(DeviceExecutionSpace{}, WithoutInitializing),                      host_view),   host_mirror_test_space(host_view));
+  check_memory_space(create_mirror_view(view_alloc(HostExecutionSpace{}),                                             host_view),   host_mirror_test_space(host_view));
+  check_memory_space(create_mirror_view(view_alloc(DeviceExecutionSpace{}, WithoutInitializing, DeviceMemorySpace{}), device_view), DeviceMemorySpace{});
+  check_memory_space(create_mirror_view(view_alloc(DeviceExecutionSpace{},                      DeviceMemorySpace{}), device_view), DeviceMemorySpace{});
+  check_memory_space(create_mirror_view(view_alloc(DeviceExecutionSpace{}, WithoutInitializing, DeviceMemorySpace{}), host_view),   DeviceMemorySpace{});
+  check_memory_space(create_mirror_view(view_alloc(DeviceExecutionSpace{},                      DeviceMemorySpace{}), host_view),   DeviceMemorySpace{});
+
+  // create_mirror_view_and_copy
+  check_memory_space(create_mirror_view_and_copy(HostMemorySpace{},   device_view), HostMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(HostMemorySpace{},   host_view),   HostMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(DeviceMemorySpace{}, device_view), DeviceMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(DeviceMemorySpace{}, host_view),   DeviceMemorySpace{});
+
+  // create_mirror_view_and_copy view_alloc
+  check_memory_space(create_mirror_view_and_copy(view_alloc(HostMemorySpace{}),   device_view), HostMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(view_alloc(HostMemorySpace{}),   host_view),   HostMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(view_alloc(DeviceMemorySpace{}), device_view), DeviceMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(view_alloc(DeviceMemorySpace{}), host_view),   DeviceMemorySpace{});
+
+  // create_mirror_view_and_copy view_alloc + execution space
+  check_memory_space(create_mirror_view_and_copy(view_alloc(HostMemorySpace{},   HostExecutionSpace{}),   device_view), HostMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(view_alloc(HostMemorySpace{},   HostExecutionSpace{}),   host_view),   HostMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(view_alloc(DeviceMemorySpace{}, DeviceExecutionSpace{}), device_view), DeviceMemorySpace{});
+  check_memory_space(create_mirror_view_and_copy(view_alloc(DeviceMemorySpace{}, DeviceExecutionSpace{}), host_view),   DeviceMemorySpace{});
+
+  // clang-format on
+}
+
+void test() {
+  Kokkos::View<int*, Kokkos::DefaultExecutionSpace> device_view("device view",
+                                                                10);
+  Kokkos::View<int*, Kokkos::HostSpace> host_view("host view", 10);
+
+  test_create_mirror_properties(device_view, host_view);
```

Comment:
If I see the code above correct its absolutely symmetric between host_view and device_view.
Couldn't we just have half as many calls above? I.e. you would use DefaultHostExecutionSpace and DefaultExecutionSpace replacing the Device/HostExecutionSpace in above code, you can get the HostMemorySpace from that too if so desired (or just use HostSpace since that is all we instantiate anyway).

And then just have here:
```
test_create_mirror_properties(device_view);
test_create_mirror_properties(host_view);
```